pkg/virt-handler/vm.go

```diff
@@ -2461,7 +2461,35 @@ func (d *VirtualMachineController) vmUpdateHelperMigrationSource(origVMI *v1.Vir
 			d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Migrating.String(), "VirtualMachineInstance is aborting migration.")
 		}
 	} else {
-		migrationConfiguration := d.clusterConfig.GetMigrationConfiguration()
+		migrationConfiguration := d.clusterConfig.GetMigrationConfiguration().DeepCopy()
+		// TODO: how to currently perform update here?
+		origVMI.Status.MigrationState.MigrationConfigSource = v1.ClusterWideConfig
+		vmi.Status.MigrationState.MigrationConfigSource = v1.ClusterWideConfig
+
+		vmiNamespace, err := d.clientset.CoreV1().Namespaces().Get(context.Background(), origVMI.Namespace, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		// Override cluster-wide migration configuration if migration policy exists
+		migrationList, err := d.clientset.MigrationPolicy().List(context.Background(), metav1.ListOptions{})
```

Comment:
use an informer rather than listing with the client here in the reconcile loop.

Reply:
Done